### PR TITLE
Make our AST great again

### PR DESCRIPTION
**Pull request by j-hui**
This is still a work-in-progress, do *NOT* merge

### CODE REVIEW

File: src/frontend/ast.mli [hunk 1]
```diff
@@ -1,156 +1,88 @@
...
-  | Lt  -> "<"
-  | Lte -> "<="
-  | And -> "&&"
-  | Or  -> "||"
-  | Concat -> "."
-  | Chord -> ","
-  | Zip -> ":"
-  | Octave -> "@"
-
-let string_of_unop o = match o with | Not -> "!" | Neg -> "-" | Sharp -> "#"
```
Comment (Contentmaudlin) [inline src/frontend/ast.mli:96]: too bad I was a huge fan of this stolen code

Comment (j-hui) [reply src/frontend/ast.mli:96]: i showed kevin and he was incredulous

File: src/frontend/parser.mly [hunk 2]
```diff
@@ -32,265 +32,268 @@
...
-  LBRACK list_lit_elements RBRACK		{ [] }
-| LBRACK RBRACK							{ [] }
+| LBRACK list_lit_elements RBRACK       { LitArray($2) }
+| LBRACK RBRACK                         { LitArray([]) }
 
 vector_lit:
-  LT list_lit_elements GT				{ [] }
-| LT GT									{ [] }
+/*| LPAREN list_lit_elements RPAREN       { LitVector($2) } */
+| LDBRACK list_lit_elements RDBRACK     { }
```
Comment (Contentmaudlin) [inline src/frontend/parser.mly]: what is `LDBRACK` or `RDBRACK`

Comment (j-hui) [reply src/frontend/parser.mly]: ignore that, it's supposed to be `[[` and `]]`, i was just sanity checking with different vector notations

**Pull request closed without merging**